### PR TITLE
Additional PHP Improvements

### PR DESCRIPTION
**Pull request by MathiasReker**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | /
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | /
| How to test?  | Everything should work as before.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11643)

### CODE REVIEW

File: classes/Customer.php [hunk 1]
```diff
@@ -239,7 +239,7 @@ public function add($autoDate = true, $nullValues = true)
         $this->id_shop_group = ($this->id_shop_group) ? $this->id_shop_group : Context::getContext()->shop->id_shop_group;
         $this->id_lang = ($this->id_lang) ? $this->id_lang : Context::getContext()->language->id;
         $this->birthday = (empty($this->years) ? $this->birthday : (int) $this->years . '-' . (int) $this->months . '-' . (int) $this->days);
-        $this->secure_key = md5(uniqid(rand(), true));
+        $this->secure_key = md5(uniqid(random_int(0, getrandmax()), true));
```
Comment (PierreRambaud) [inline classes/Customer.php]: random_int is available since PHP7, prefer mt_rand instead

Comment (MathiasReker) [reply classes/Customer.php]: Thanks. Solved.

Comment (MathiasReker) [reply classes/Customer.php]: Thanks! Solved.

File: controllers/admin/AdminShopUrlController.php [hunk 2]
```diff
@@ -493,7 +493,7 @@ public function processUpdate()
     {
         $this->redirect_shop_url = false;
         $current_url = parse_url($_SERVER['REQUEST_URI']);
-        if (trim(dirname(dirname($current_url['path'])), '/') == trim($this->object->getBaseURI(), '/')) {
+        if (trim(dirname($current_url['path'],2), '/') == trim($this->object->getBaseURI(), '/')) {
```
Comment (123monsite-regis) [inline controllers/admin/AdminShopUrlController.php]: sorry , `level` params of `dirname` function was added on php 7.0;
http://php.net/manual/fr/function.dirname.php
i does not know if php 7 will be required for ps , can please ps team answer about the php minimum version ?

Comment (MathiasReker) [reply controllers/admin/AdminShopUrlController.php]: Nice catch, thanks.

**Review by mickaelandrieu — APPROVED** [review]

**Comment by mickaelandrieu** [comment]
ping @PierreRambaud, except if I'm wrong this can't be validated by QA team: LGTM?

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
@mickaelandrieu Agreed with you!

**Comment by mickaelandrieu** [comment]
So be it!

**Pull request merged**